pom.xml

```diff
@@ -57,7 +57,7 @@
       <dependency>
           <groupId>commons-httpclient</groupId>
           <artifactId>commons-httpclient</artifactId>
-          <version>3.1</version>
+          <version>3.1-jenkins-1</version>
```

Comment:
I don't see a reference to "-jenkins-1" in the [maven repository listing](https://mvnrepository.com/artifact/commons-httpclient/commons-httpclient).  I assumed this would change to a reference to the Jenkins plugin which provides that library.  Does this compile successfully on a machine that is not your primary development machine?

Reply:
Yeah, about that. I can download the dependency from  http://repo.jenkins-ci.org/public/commons-httpclient/commons-httpclient/3.1-jenkins-1/. 

The problem is it still doesn't contain the class we need. It seems to me it's because the class is in [src/contrib subfolder in the repository](https://github.com/jenkinsci/lib-commons-httpclient/tree/commons-httpclient-3.1-jenkins-1/src/contrib/org/apache/commons/httpclient/contrib) and is not packaged at all. 

Do you have any idea what's going on?

Reply:
I suspect that means they are also not packaging it as part of the commons-httpclient.  That may leave us with no choice but to add it to the git client plugin and openly admit that it is part of the published API of the git client plugin.

Reply:
Tried filling  a PR at https://github.com/jenkinsci/lib-commons-httpclient/pull/1